### PR TITLE
Update build number and add libcurl dep

### LINKED ISSUE #34: xerces-c 3.2.4 and libcurl
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

It appears that the xerces-c 3.2.4 build is now picking up more libraries than 3.2.3 was (including some from the system):

Xerces-c 3.2.4 (on a system that has the libraries):
```
$> ldd libxerces-c-3.2.so 
	linux-vdso.so.1 =>  (0x00007fffef1e2000)
	libnsl.so.3 => /opt/conda/envs/mini-check/lib/././libnsl.so.3 (0x00007fc619dfe000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fc61963b000)
	libcurl.so.4 => /lib64/libcurl.so.4 (0x00007fc6193d1000)
	libicui18n.so.70 => /opt/conda/envs/mini-check/lib/././libicui18n.so.70 (0x00007fc6190a7000)
	libicuuc.so.70 => /opt/conda/envs/mini-check/lib/././libicuuc.so.70 (0x00007fc618ea5000)
	libicudata.so.70 => /opt/conda/envs/mini-check/lib/././libicudata.so.70 (0x00007fc617288000)
	libstdc++.so.6 => /opt/conda/envs/mini-check/lib/././libstdc++.so.6 (0x00007fc619c45000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fc616f86000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fc616bb8000)
	libgcc_s.so.1 => /opt/conda/envs/mini-check/lib/././libgcc_s.so.1 (0x00007fc619c2b000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fc619be6000)
	libidn.so.11 => /lib64/libidn.so.11 (0x00007fc616985000)
	libssh2.so.1 => /lib64/libssh2.so.1 (0x00007fc616758000)
	libssl3.so => /lib64/libssl3.so (0x00007fc6164f3000)
	libsmime3.so => /lib64/libsmime3.so (0x00007fc6162cb000)
	libnss3.so => /lib64/libnss3.so (0x00007fc615f91000)
	libnssutil3.so => /lib64/libnssutil3.so (0x00007fc615d61000)
	libplds4.so => /lib64/libplds4.so (0x00007fc615b5d000)
	libplc4.so => /lib64/libplc4.so (0x00007fc615958000)
	libnspr4.so => /lib64/libnspr4.so (0x00007fc615719000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fc615515000)
	libgssapi_krb5.so.2 => /lib64/libgssapi_krb5.so.2 (0x00007fc6152c8000)
	libkrb5.so.3 => /lib64/libkrb5.so.3 (0x00007fc614fdf000)
	libk5crypto.so.3 => /lib64/libk5crypto.so.3 (0x00007fc614dac000)
	libcom_err.so.2 => /lib64/libcom_err.so.2 (0x00007fc614ba8000)
	liblber-2.4.so.2 => /lib64/liblber-2.4.so.2 (0x00007fc614999000)
	libldap-2.4.so.2 => /lib64/libldap-2.4.so.2 (0x00007fc614744000)
	libz.so.1 => /lib64/libz.so.1 (0x00007fc61452e000)
	libssl.so.10 => /lib64/libssl.so.10 (0x00007fc6142bc000)
	libcrypto.so.10 => /lib64/libcrypto.so.10 (0x00007fc613e59000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fc613c51000)
	libkrb5support.so.0 => /lib64/libkrb5support.so.0 (0x00007fc613a41000)
	libkeyutils.so.1 => /lib64/libkeyutils.so.1 (0x00007fc61383d000)
	libresolv.so.2 => /lib64/libresolv.so.2 (0x00007fc613623000)
	libsasl2.so.3 => /lib64/libsasl2.so.3 (0x00007fc613406000)
	libselinux.so.1 => /lib64/libselinux.so.1 (0x00007fc6131df000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007fc612fa8000)
	libpcre.so.1 => /lib64/libpcre.so.1 (0x00007fc612d46000)
	libfreebl3.so => /lib64/libfreebl3.so (0x00007fc612b43000)
```

Xerces-c 3.2.3 (build 5):
```
$> ldd libxerces-c-3.2.so 
	linux-vdso.so.1 =>  (0x00007ffdf370c000)
	libnsl.so.3 => /opt/conda/envs/mini-check-orig/lib/././libnsl.so.3 (0x00007fd8c4bf8000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fd8c4434000)
	libicui18n.so.70 => /opt/conda/envs/mini-check-orig/lib/././libicui18n.so.70 (0x00007fd8c410a000)
	libicuuc.so.70 => /opt/conda/envs/mini-check-orig/lib/././libicuuc.so.70 (0x00007fd8c3f08000)
	libicudata.so.70 => /opt/conda/envs/mini-check-orig/lib/././libicudata.so.70 (0x00007fd8c22eb000)
	libstdc++.so.6 => /opt/conda/envs/mini-check-orig/lib/././libstdc++.so.6 (0x00007fd8c4a3f000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fd8c1fe9000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fd8c1c1b000)
	libgcc_s.so.1 => /opt/conda/envs/mini-check-orig/lib/././libgcc_s.so.1 (0x00007fd8c4a25000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fd8c49e0000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fd8c1a17000)
```

This is actually resulting in build issues for a package I'm trying to build on a stripped down system right now:
```
/<...>/ld: warning: libcurl.so.4, needed by /<...>placehold_placehol/lib/libxerces-c.so, not found (try using -rpath or -rpath-link)
/<...>/libxerces-c.so: undefined reference to `curl_easy_getinfo'
/<...>/libxerces-c.so: undefined reference to `curl_easy_init'
/<...>/libxerces-c.so: undefined reference to `curl_slist_append'
/<...>/libxerces-c.so: undefined reference to `curl_multi_add_handle'
/<...>/libxerces-c.so: undefined reference to `curl_multi_fdset'
<...>
```

I assume curl is being built against because the --disable-network flag was removed from the recipe and it is now building against the curl accessor. This might not have been built against previously because curl wasn't being found for some reason (so the accessor wasn't building) or maybe due to the curl m4 changes in xerces-c 3.2.4. 

Either way, I assume the fix for that library is to either add --disable-network back in or to add libcurl to the host requirements and us --with-curl. Though, there may or may not be issues with the other libraries coming from the system (libcrypt, libcrypto, etc.). The current build number 0 version will also likely need to allow people to pick up the version with additional requirements without issues.

I'm building the Linux package for xerces-c to check the fix mentioned above and will submit a PR if this resolves things.

### Installed packages

```shell
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
icu                       70.1                 h27087fc_0    conda-forge
libgcc-ng                 12.2.0              h65d4601_18    conda-forge
libgomp                   12.2.0              h65d4601_18    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libstdcxx-ng              12.2.0              h46fd767_18    conda-forge
xerces-c                  3.2.4                h55805fa_0    conda-forge
```

### Environment info

```shell
active environment : mini-check
    active env location : /opt/conda/envs/mini-check
            shell level : 2
       user config file : /root/.condarc
 populated config files : /root/.condarc
          conda version : 22.9.0
    conda-build version : 3.22.0
         python version : 3.9.12.final.0
       virtual packages : __linux=4.15.0=0
                          __glibc=2.17=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /opt/conda  (writable)
      conda av data dir : /opt/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/conda/pkgs
                          /root/.conda/pkgs
       envs directories : /opt/conda/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/22.9.0 requests/2.27.1 CPython/3.9.12 Linux/4.15.0-159-generic centos/7.9.2009 glibc/2.17
                UID:GID : 0:0
             netrc file : None
           offline mode : False
```

### PR DESCRIPTION
**Pull request by Marie-Terrell**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Closes #34 

I've tried, but am unable to run the build locally without the make checks failing (before and after this change), so I am at least putting the proposed change here. Please feel free to close this and re-open a new fix for #34 if this does not fix the issue at hand or for the incorrect submission (not being built beforehand).

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ requirements:
   host:
     - icu       # [not win]
     - libnsl  # [linux]
+    - libcurl
```
Comment (gillins) [inline recipe/meta.yaml:31]: I think this also needs to be added to the `run` section...

Comment (gillins) [reply recipe/meta.yaml:31]: And both `host` and `run` need to be `[not win]`?

Comment (Marie-Terrell) [reply recipe/meta.yaml]: Thanks! Updated!

Comment (gillins) [reply recipe/meta.yaml]: Can you fix this linting error by inserting another space before the `#`

**Comment by gillins** [comment]
Thanks!

**Pull request merged**